cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -979,23 +981,22 @@ ExtensionIdRegistry::SubstraitAggregateToArrow DecodeBasicAggregate(
             options = std::make_shared<compute::VarianceOptions>(ddof);
           }
         }
-        fixed_arrow_func += arrow_function_name;
 
-        ARROW_ASSIGN_OR_RAISE(compute::Expression arg, call.GetValueArg(0));
-        const FieldRef* arg_ref = arg.field_ref();
-        if (!arg_ref) {
-          return Status::Invalid("Expected an aggregate call ", call.id().uri, "#",
-                                 call.id().name, " to have a direct reference");
+        fixed_arrow_func += arrow_function_name;
+        std::vector<FieldRef> target;
+        for (int i = 0; i < call.size(); i++) {
+          ARROW_ASSIGN_OR_RAISE(compute::Expression arg, call.GetValueArg(i));
+          const FieldRef* arg_ref = arg.field_ref();
```

Comment:
Remove `const` so it can be moved below.

Reply:
@felipecrv My understanding is with `const FieldRef*` then the move below has no effect / need to copy the data (because it is pointer to const FieldRef) and with `FieldRef*` the move operator can then actually move the object and not copying it?

Reply:
You're right. This is one of the many ways copies can creep in.

But looking at the definition of `field_ref` I noticed it can only return `const`, so removing `const` won't really be possible. So I recommend removing the `std::move(*arg_ref)` below so it doesn't give the wrong impression that a move is happening.

Reply:
More details on the move const issue: https://stackoverflow.com/questions/27810535/why-does-calling-stdmove-on-a-const-object-call-the-copy-constructor-when-pass

Reply:
Thanks! Removed `std::move(*arg_ref)`